extensions/amp-fx-parallax/0.1/amp-fx-parallax.js

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Observable} from '../../../src/observable';
+import {getService} from '../../../src/service';
+import {isExperimentOn} from '../../../src/experiments';
+import {onDocumentReady} from '../../../src/document-ready';
+import {setStyles} from '../../../src/style';
+import {toArray} from '../../../src/types';
+import {user} from '../../../src/log';
+import {viewportForDoc} from '../../../src/viewport';
+import {vsyncFor} from '../../../src/vsync';
+
+const ATTR = 'amp-fx-parallax';
+const EXPERIMENT = ATTR;
+
+/**
+ * Installs parallax handlers, tracks the previous scroll position and
+ * implements post-parallax-update scroll hooks.
+ */
+export class ParallaxService {
```

Comment:
let's move the service under `service` folder and make it an "ampdoc" level service rather than a window service. See  [video-manager.js](https://github.com/ampproject/amphtml/blob/master/src/video-manager.js) and [video-manager-impl](https://github.com/ampproject/amphtml/blob/master/src/service/video-manager-impl.js) for an example.

Note that moving this outside of the component does NOT mean the core AMP library (`v0.js`) grows in size because it would only get bounded into binaries that actually use it ( e.g. amp-fx-parallax/0.1/amp-fx-parallax.js)